src/test/util_string_tests.cpp

```diff
@@ -11,13 +11,25 @@ using namespace util;
 
 BOOST_AUTO_TEST_SUITE(util_string_tests)
 
+template <unsigned NumArgs>
+void TfmFormatZeroes(const char* fmt)
+{
+    std::apply([fmt](auto... args) {
+        (void)tfm::format(std::string{fmt}, args...);
+    }, std::tuple_cat(std::array<int, NumArgs>{}));
+}
+
 // Helper to allow compile-time sanity checks while providing the number of
 // args directly. Normally PassFmt<sizeof...(Args)> would be used.
 template <unsigned NumArgs>
 inline void PassFmt(ConstevalFormatString<NumArgs> fmt)
 {
     // Execute compile-time check again at run-time to get code coverage stats
     util::detail::CheckNumFormatSpecifiers<NumArgs>(fmt.fmt);
+
+    // Prove parity with tinyformat, which will throw if the number of
```

Comment:
In commit "test: Prove+document ConstevalFormatString/tinyformat parity" (4352596f4167d3e0c1d39ee46c6d5096f0228c61)

Not sure this is really parity, since it only shows that tinyformat doesn't throw when ConstevalFormatString doesn't throw, not that tinyformat throws when ConstevalFormatString does (which we probably don't care about). A more literal comment like "Make sure tinyformat doesn't throw if ConstevalFormat didn't throw" might make it more obvious what benefit of this check is.

Reply:
Updated with more verbose comment, let me know if you think it's okay.

```C++
// If ConstevalFormatString didn't throw above, make sure tinyformat doesn't
// throw either for the same format string and parameter count combination.
// Proves that we have some extent of protection from runtime errors
// (tinyformat may still throw for some type mismatches).
```